### PR TITLE
Adds test and build workflow for windows

### LINKED ISSUE #51: Add Windows Support
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/95, add Windows support.

- [ ] Passing CI on Windows
- [ ] Documentation

### PR DESCRIPTION
**Pull request by eirsep**
Signed-off-by: Surya Sashank Nistala <snistala@amazon.com>

### Description
Adds test and build workflow for windows OS
 
### Issues Resolved
#51
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -89,5 +89,37 @@ jobs:
...
+    steps:
+      # This step uses the setup-java Github action: https://github.com/actions/setup-java
+      - name: Set Up JDK 14
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      # This step uses the checkout Github action: https://github.com/actions/checkout
+      - name: Checkout Branch
+        uses: actions/checkout@v2
+      # This is a hack, but this step creates a link to the X: mounted drive, which makes the path short enough to work on Windows
```
Comment (dblock) [inline .github/workflows/build.yml:108]: What do you get without this?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -89,5 +89,33 @@ jobs:
       - name: Upload Artifacts
         uses: actions/upload-artifact@v1
         with:
-          name: asynchronous-search-plugin
+          name: asynchronous-search-plugin-ubuntu
+          path: asynchronous-search-artifacts
+  windows-build:
+    # Job name
+    name: Build Asynchronous Search
+    # This job runs on Linux.
```
Comment (Bukhtawar) [inline .github/workflows/build.yml:97]: comment?

Comment (eirsep) [reply .github/workflows/build.yml]: fixed this

File: .github/workflows/build.yml (hunk 1 above)
Comment (eirsep) [reply .github/workflows/build.yml]: removed it

**Review by Bukhtawar — APPROVED** [review]

**Review by anshul291995 — APPROVED** [review]

**Review by mitalawachat — APPROVED** [review]

**Pull request merged**